rosa_planning/rosa-understanding-terraform.adoc

```diff
@@ -0,0 +1,163 @@
+:_content-type: ASSEMBLY
+include::_attributes/attributes-openshift-dedicated.adoc[]
+
+[id="rosa-understanding-terraform"]
+= Preparing Terraform to install ROSA clusters
+:context: rosa-understanding-terraform
+
+toc::[]
+
+Terraform is an infrastructure-as-code tool that provides a way to configure your resources once and replicate those resources as desired. Terraform accomplishes the creation tasks by using a declarative language. You declare what you want the final state of the infrastructure resource to be, and Terraform creates these resources to your specifications.
+
+:FeatureName: Red Hat Cloud Services Terraform Provider 
+include::snippets/technology-preview.adoc[]
+
+[id="sd-terraform-prereqs"]
+== Prequisites
+ 
+To use link:https://registry.terraform.io/providers/terraform-redhat/rhcs/latest/docs[the Red Hat Cloud Services provider] inside your Terraform configuration, you must meet the following prerequisites:
+
+* You completed link:https://console.redhat.com/openshift/create/rosa/getstarted[the ROSA getting started] requirements
++
+You must complete AWS account configuration and local configuration before creating ROSA clusters.
+* An offline link:https://console.redhat.com/openshift/token/rosa[OCM token]
++
+This token is generated through the Red Hat Hybrid Cloud Console. It is unique to your account and should not be shared. The token is generated based off your account access and permissions.
+
+* link:https://go.dev/doc/install[GoLang version 1.20] or newer
++
+You must have the latest version of Go installed and usable on your system to build components with Terraform.
+
+* link:https://developer.hashicorp.com/terraform/downloads[Terraform version 1.4.6] or newer.
++
+You must have Terraform configured for your local system. The Terraform website contains installation options for MacOS, Windows, and Linux.
+
+* You must create your AWS account roles
++
+You must create your AWS account-wide roles. These roles can be created using Terraform. For more information, see the Account roles Terraform example.
++
+The specific account-wide IAM roles and policies provide the STS permissions required for ROSA support, installation, control plane, and compute functionality. This includes account-wide Operator policies. See the Additional resources for more information on the AWS account roles.
++
+To create the account roles using Terraform, see the Account roles Terraform example.
+
+* AWS permissions
++
+The following excerpt lists the minimum AWS permissions required to run Terraform.
++
+[%collapsible]
```

Comment:
Good call making this collapsible. Suggestion to change the "Details" label to something more descriptive.